crates/sparrow-main/tests/e2e/basic_error_tests.rs

```diff
@@ -204,7 +204,7 @@ async fn test_illegal_cast_to_generic() {
           - "  --> Query:1:19"
           - "  |"
           - "1 | { m: Numbers.m as number }"
-          - "  |                   ^^^^^^ Unable to cast to type 'number'"
+          - "  |                   ^^^^^^ Unable to cast to type 'num'"
```

Comment:
This is obviously wrong. But shows that the user can write the same name as our built-in constraints. It's hard to determine whether it's a valid or invalid use of the string though, so I'm not sure what the best solution is for now. Perhaps just write out the full name of the constraint in the `Display` for the `TypeParameter`, and we work on better error reporting later.